common/src/types/v0/store/node.rs

```diff
@@ -37,20 +37,31 @@ pub struct NodeSpec {
     /// Cordon labels.
     #[serde(default)] // Ensure backwards compatibility in etcd when upgrading.
     cordon_labels: Vec<String>,
+    /// Drain labels.
+    #[serde(default)] // Ensure backwards compatibility in etcd when upgrading.
+    drain_labels: Vec<String>,
+    /// Drain state
+    #[serde(default)] // Ensure backwards compatibility in etcd when upgrading.
+    is_draining: bool,
 }
+
 impl NodeSpec {
     /// Return a new `Self`
     pub fn new(
         id: NodeId,
         endpoint: String,
         labels: NodeLabels,
         cordon_label: Option<Vec<String>>,
+        drain_label: Option<Vec<String>>,
+        is_draining: Option<bool>,
```

Comment:
is_draining is needed to distinguish between a node which is draining and one in which the drain is completed.
Both will still have at least one drain label.
When the reconciler is implemented, it will set the is_draining flag to false.

Reply:
Okay. Thanks